src/nomad_simulations/properties/band_structure.py

```diff
@@ -0,0 +1,370 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from structlog.stdlib import BoundLogger
+from typing import Optional, Tuple
+import pint
+
+from nomad.metainfo import Quantity, Section, Context, SubSection
+
+from nomad_simulations.numerical_settings import KSpace
+from nomad_simulations.physical_property import PhysicalProperty
+from nomad_simulations.properties import ElectronicBandGap, FermiSurface
+from nomad_simulations.utils import get_sibling_section
+
+
+class BaseElectronicEigenvalues(PhysicalProperty):
+    """
+    A base section used to define basic quantities for the `ElectronicEigenvalues`, `FermiSurface`, and
+    `ElectronicBandStructure` properties. This section serves to define `FermiSurface` and without needing to specify
+    other quantities that appear in `ElectronicEigenvalues`
+    """
+
+    iri = ''
+
+    n_bands = Quantity(
+        type=np.int32,
+        description="""
+        Number of bands / eigenvalues.
+        """,
+    )
+
+    value = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        Value of the electronic eigenvalues.
+        """,
+    )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        # ! `n_bands` need to be set up during initialization of the class
+        self.rank = [self.n_bands]
```

Comment:
or maybe get it from kwargs

Reply:
Yeah, sorry, at this point I do not care too much about this; at the end, this is a definition, so it should go in m_def. 

I am waiting for Markus to implement something on this direction, so my point is: don't pay too much attention to this dirty assignment for now 😄

Reply:
I was just concerned about the same error you were having previously that you have to convert it to int when read from msection

Reply:
Yes, I had to, thanks for catching it up. In the new version you can see I did `self.rank = [int(self.n_bands)]`, but I will change it 👍🏻